openedx/core/djangoapps/content/course_overviews/tasks.py

```diff
@@ -0,0 +1,47 @@
+import logging
+
+from celery import task
+from celery_utils.logged_task import LoggedTask
+from celery_utils.persist_on_failure import PersistOnFailureTask
+from django.conf import settings
+
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+from xmodule.modulestore.django import modulestore
+
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+
+log = logging.getLogger(__name__)
+
+
+def chunks(sequence, chunk_size):
+    return (sequence[index: index + chunk_size] for index in xrange(0, len(sequence), chunk_size))
+
+
+class GenerateCourseOverviewException(Exception):
+    pass
+
+
+class _BaseTask(PersistOnFailureTask, LoggedTask):  # pylint: disable=abstract-method
+    """
+    Include persistence features, as well as logging of task invocation.
+    """
+    abstract = True
+
+
+@task(base=_BaseTask, routing_key=settings.HIGH_MEM_QUEUE)
+def generate_course_overview(*args, **kwargs):
+    if kwargs['all']:
+        course_keys = [course.id for course in modulestore().get_course_summaries()]
+    else:
+        try:
+            course_keys = [CourseKey.from_string(arg) for arg in args]
+        except InvalidKeyError:
+            raise GenerateCourseOverviewException('Invalid key specified.')
+
+    chunk_size = kwargs['chunk_size']
+    force_update = kwargs['force_update']
+
+    for course_key_group in chunks(course_keys, chunk_size):
```

Comment:
This doesn't create subtasks for each chunk.  So I imagine you'll still run into the same memory management issue - since we're still processing all courses within the same process.

Reply:
Haha, whoops!  Thank you.